aws-deploy/tasks/main.yml

```diff
@@ -1,25 +1,28 @@
-- name: create launch configuration
-  ec2_lc:
-    region: "{{ region }}"
-    name: "{{ launch_config_name }}"
-    image_id: "{{ image_id }}"
-    key_name: "{{ keypair }}"
-    instance_type: "{{ instance_type }}"
-    security_groups: "{{ security_groups }}"
-    instance_monitoring: yes
-    instance_profile_name: "{{ instance_profile_name }}"
-
-- name: Configure Auto Scaling Group and perform rolling deploy
-  ec2_asg:
-    region: "{{ region }}"
-    name: "{{ auto_scaling_group_name }}"
-    launch_config_name: "{{ launch_config_name }}"
-    health_check_type: ELB
-    health_check_period: 300
-    desired_capacity: "{{ desired_capacity | default(2) }}"
-    min_size: "{{ min_size | default(2) }}"
-    max_size: "{{ max_size | default(2) }}"
-    replace_all_instances: yes
-    replace_batch_size: "{{ replace_batch_size | default(1) }}"
-    state: present
-  register: asg_result
+- name: create launch configuration
+  ec2_lc:
+    region: "{{ region }}"
+    name: "{{ launch_config_name }}"
+    image_id: "{{ image_id }}"
+    key_name: "{{ keypair }}"
+    instance_type: "{{ instance_type }}"
+    security_groups: "{{ security_groups }}"
+    instance_monitoring: yes
+    instance_profile_name: "{{ instance_profile_name }}"
+
+- name: Configure Auto Scaling Group and perform rolling deploy
+  ec2_asg:
+    desired_capacity: "{{ desired_capacity | default(2) }}"
+    health_check_period: 300
+    health_check_type: ELB
+    launch_config_name: "{{ launch_config_name }}"
+    load_balancers: "{{ app_loadbalancers }}"
+    max_size: "{{ max_size | default(2) }}"
+    min_size: "{{ min_size | default(2) }}"
+    name: "{{ auto_scaling_group_name }}"
+    region: "{{ region }}"
+    replace_all_instances: yes
+    replace_batch_size: "{{ replace_batch_size | default(1) }}"
+    state: present
+    tags: "{{ app_tags | default(null) }}"
```

Comment:
Just found this:

http://stackoverflow.com/questions/33394333/how-to-add-a-key-value-to-a-dict-in-an-ansible-playbook-conditional

Ick...seems like this isn't something easily done.

IMO, we should have only one and we should *always* specify all the options. The mantra in ansible is to make the playbooks idempotent so setup and update should look the same. That's why it's declarative "roles" instead of imperative actions.

If you look at the convention I've been toying with here:
  https://github.com/department-of-veterans-affairs/devops/blob/master/ansible/roles/vetsgov/aws-config/defaults/main.yml

Basically there is one role that has *all* of the site configuration as a static key/value pair. This role is then loaded via meta/main.yml first in pretty much every other role which lets us just reference stuff via `aws_config[env].subnet_ids`.

In that work, rather than aws-deploy, you'd name the role something like aws-frontend-stack, and the goal is to always ensure the whole lb/asg/lc/ami/sg setup was in sync with whatever was specified in aws_config.

Deploy then becomes, update aws_config. Commit. Run play. This creates a nice audit log....

Thoughts?